### DISCUSSION TITLE
Meld shows errors at launch when installed from Scoop

### DISCUSSION THREAD

**Discussion by ZaLiTHkA** [ScoopInstaller/Extras#13703] [Q&A] [answered]
running under Windows 11 Pro 23H2, when I install Meld (`3.22.2` as of writing this) through Scoop, I get a series of error messages at launch:

- The code execution cannot proceed because libglib-2.0-0.dll was not found. Reinstalling the program may fix this problem.
- same for libgobject-2.0-0.dll
- same for libintl-8.dll
- same for libgio-2.0-0.dll

but then Meld opens just fine and I am able to do file and folder comparisons with no issues, at least none that I have found yet. just for clarification, all of the files mentioned in these error popups do exist in my `%USERPROFILE%\scoop\apps\meld\3.22.2` folder.

for comparison, when I download the exact same `msi` file and install it manually, then Meld starts up without these errors.

just wondering if anybody else is also seeing this behaviour?

**Comment by HUMORCE** [comment]
I have installed Meld before, and it seems to work fine.

![Screenshot_240727071004](https://github.com/user-attachments/assets/c1b33826-62cf-42a0-91e4-292cd0672b34)

**Reply by ZaLiTHkA** [reply]
interesting.. I see Meld is still on the same version, but Scoop itself has had two updates since I posted this.

I will try this again on my Windows 11 machine again in the morning and let you know what happens.

**Reply by ZaLiTHkA** [reply]
sorry to say, I completely forgot about this one.. a comment below a few days back pinged me. tl;dr, 27/07 is my birthday, and then life happened, and then I switched my main desktop environment to Linux. 🙃

**Comment by needleshaped** [comment]
EDIT: WORKAROUNDED

I do have same problem appeared (N.B. right after installation there were no errors, it's just I don't use Meld everyday, so no info of possible trigger, apart from autoupdate).

![image](https://github.com/user-attachments/assets/0e212e88-b699-48a0-a480-fc5c9cd19030)

Writing investigation results here:

1. Error comes from gdbus.exe
1. Meld is run by shortcut with 
    - Target:  C:\scoop\apps\meld\current\Meld.exe
    - Start in: C:\scoop\apps\meld\current
1. c:\scoop\apps\meld\current\ **does** contains libs with complaint:
    ```
    etc\
    lib\
    share\
    libatk-1.0-0.dll
    libbrotlicommon.dll
    libbrotlidec.dll
    libbz2-1.dll
    libcairo-2.dll
    libcairo-gobject-2.dll
    libcrypto-3-x64.dll
    libdatrie-1.dll
    libepoxy-0.dll
    libexpat-1.dll
    libffi-8.dll
    libfontconfig-1.dll
    libfreetype-6.dll
    libfribidi-0.dll
    libgcc_s_seh-1.dll
    libgdk_pixbuf-2.0-0.dll
    libgdk-3-0.dll
    libgio-2.0-0.dll
    libgirepository-1.0-1.dll
    libglib-2.0-0.dll
    libgmodule-2.0-0.dll
    libgobject-2.0-0.dll
    libgraphite2.dll
    libgtk-3-0.dll
    libgtksourceview-4-0.dll
    libharfbuzz-0.dll
    libiconv-2.dll
    libintl-8.dll
    liblzma-5.dll
    libmpdec-4.dll
    libpango-1.0-0.dll
    libpangocairo-1.0-0.dll
    libpangoft2-1.0-0.dll
    libpangowin32-1.0-0.dll
    libpcre2-8-0.dll
    libpixman-1-0.dll
    libpng16-16.dll
    libpython3.11.dll
    librsvg-2-2.dll
    libssl-3-x64.dll
    libstdc++-6.dll
    libthai-0.dll
    libwinpthread-1.dll
    libxml2-2.dll
    zlib1.dll
    Meld.exe
    MeldConsole.exe
    install.json
    manifest.json
    frozen_application_license.txt
    ```
1. scoop uninstall meld; scoop uninstall meld - no change, starts with errors
1. gdbus.exe is located in (!) c:\scoop\apps\meld\current\lib\
1. Copying 4 above libraries to c:\scoop\apps\meld\current\lib\ results in another 4 errors about other libs (it seems there is limit to errors shown).
1. ✅ Copying ALL libraries (*.lib) from c:\scoop\apps\meld\current\ to c:\scoop\apps\meld\current\lib\ DOES HELP, NO MORE ERRORS. Seems gdbus.exe cannot locate libraries in above folder. I wonder, if the directory structure is intended and/or unchanged.
1. ✅ Just removing gdbus.exe ALSO HELPS 

P.S. meld.json: https://github.com/idleberg/scoop-extras/commits/master/meld.json

**Reply by nicriverhoo** [reply]
just copy meld/3.22.2/lib/gdbus.exe to meld/3.22.2/gdbus.exe also helps.

**Reply by joatansampaio** [reply]
> just copy meld/3.22.2/lib/gdbus.exe to meld/3.22.2/gdbus.exe also helps.

That fixed my problem. Thanks!!

**Reply by haussmann** [reply]
Relates to Bug: https://github.com/ScoopInstaller/Extras/issues/16156

Thank you very much for your investigation and tests. I implemented the workaround here: https://github.com/ScoopInstaller/Extras/pull/16992

**Reply by haussmann** [reply, marked as answer]
The workaround is now merged. Please let us know if the error still occurs after `scoop update extras/meld -f`

**Reply by z-Fng** [reply]
- https://gitlab.gnome.org/GNOME/meld/-/issues/886
- https://gitlab.gnome.org/GNOME/meld/-/issues/940
